protocol/testing/e2e/gov/add_new_market_test.go

```diff
@@ -0,0 +1,256 @@
+						pricestest.WithId(5),
+					).Param,
+				},
+				&perptypes.MsgCreatePerpetual{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Params: perptest.GeneratePerpetual(
+						perptest.WithId(5),
+					).Params,
+				},
+				&clobtypes.MsgCreateClobPair{
+					Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					ClobPair: *clobtest.GenerateClobPair(
+						clobtest.WithId(5),
+					),
+				},
+				&delaymsgtypes.MsgDelayMessage{
+					Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+					Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive),
+					DelayBlocks: 10,
+				},
+			},
+			expectSubmitProposalFail: true,
+			updateClobDelayBlocks:    10,
+			expectedProposalStatus:   govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+		},
+		// TODO(CORE-584): Uncomment this case. Currently gov EndBlocker doesn't recover from panic in
+		// message handler, and invalid message signer results in panic. We should fix this behavior,
+		// and verifies that the proposal just fails instead.
+		// "Invalid signer on `MsgDelayMessage`": {
+		// 	proposedMsgs: []sdk.Msg{
+		// 		&pricestypes.MsgCreateOracleMarket{
+		// 			Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		// 			Params:    testMarketParam1001.Param,
+		// 		},
+		// 		&perptypes.MsgCreatePerpetual{
+		// 			Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		// 			Params:    testPerpetual1001.Params,
+		// 		},
+		// 		&clobtypes.MsgCreateClobPair{
+		// 			Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		// 			ClobPair:  *testClobPair1001,
+		// 		},
+		// 		&delaymsgtypes.MsgDelayMessage{
+		// 			Authority:   authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		// 			Msg:         encoding.EncodeMessageToAny(t, msgUpdateClobPair1001ToActive_WrongAuthority),
+		// 			DelayBlocks: 10,
+		// 		},
+		// 	},
+		// 	updateClobDelayBlocks: 10,
+		//  expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+		// },
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			tApp := testapp.NewTestAppBuilder().WithGenesisDocFn(func() (genesis types.GenesisDoc) {
+				genesis = testapp.DefaultGenesis()
+				testapp.UpdateGenesisDocWithAppStateForModule(
+					&genesis,
+					func(genesisState *govtypesv1.GenesisState) {
+						genesisState.Params.VotingPeriod = &testapp.TestVotingPeriod
+					},
+				)
+				return genesis
+			}).WithTesting(t).Build()
+			ctx := tApp.InitChain()
+
+			initMarketParams := tApp.App.PricesKeeper.GetAllMarketParams(ctx)
+			initPerpetuals := tApp.App.PerpetualsKeeper.GetAllPerpetuals(ctx)
+			initClobPairs := tApp.App.ClobKeeper.GetAllClobPairs(ctx)
+
+			ctx = testapp.SubmitAndTallyProposal(
+				t,
+				ctx,
+				&tApp,
+				tc.proposedMsgs,
+				tc.expectSubmitProposalFail,
+				tc.expectedProposalStatus,
+			)
+
+			if tc.expectSubmitProposalFail {
+				return
```

Comment:
if the proposal submission fails, are there conditions that we assert here?

like there's no pending gov proposal, etc